contracts/Vault1.sol

```diff
@@ -1,6 +1,27 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+import "../openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+
+
+/// @notice Vault to deposit and withdraw any ERC20 token.
 contract Vault1 {
-    
+    address token;
```

Comment:
We don't need to store this separately since we are already storing the contract. You can always get address using address(token_contract)